improver/lightning.py

```diff
@@ -204,3 +204,220 @@ def latitude_to_threshold(
+        cin = cubes.extract(
+            iris.Constraint(
+                cube_func=lambda cube: "atmosphere_specific_convective_inhibition"
+                in cube.name()
+            )
+        )
+        if cin:
+            cin = cin.merge_cube()
+        else:
+            raise ValueError(
+                f"No cube named atmosphere_specific_convective_inhibition found "
+                f"in {cubes}"
+            )
+        apcp = cubes.extract(
+            iris.Constraint(
+                cube_func=lambda cube: "precipitation_amount" in cube.name()
+            )
+        )
+        if apcp:
+            apcp = apcp.merge_cube()
+        else:
+            raise ValueError(f"No cube named precipitation_amount found in {cubes}")
+
+        (cape_time,) = list(cape.coord("time").cells())
+        (apcp_time,) = list(apcp.coord("time").cells())
+        if cape_time.point != apcp_time.bound[0]:
+            raise ValueError(
+                f"CAPE cube time ({cape_time.point}) should be valid at the "
+                f"precipitation_accumulation cube lower bound ({apcp_time.bound[0]})."
+            )
+        if np.diff(apcp_time.bound) not in [timedelta(hours=3)]:
+            raise ValueError(
+                f"Precipitation_accumulation cube time window must be three hours, "
+                f"not {np.diff(apcp_time.bound)}."
+            )
+        # Following time/space checks should be made for all diagnostics
+        if cape.coord("forecast_reference_time") != apcp.coord(
+            "forecast_reference_time"
+        ):
+            raise ValueError(
+                "Supplied cubes must have the same forecast reference times"
+            )
+        if not spatial_coords_match([cape, apcp]):
+            raise ValueError("Supplied cubes do not have the same spatial coordinates")
+
+        return cape, liftidx, pwat, cin, apcp
+
+    def process(self, cubes: CubeList, model_id_attr: str = None) -> Cube:
+        """
+        From the supplied CAPE, LIFTIDX, PWAT, CIN, APCP cubes, calculate a probability
+        of lightning cube.
+
+        Args:
+            cubes:
+                Cubes of CAPE, LIFTIDX, PWAT, CIN, APCP.
+            model_id_attr:
+                The name of the dataset attribute to be used to identify the source
+                model when blending data from different models.
+
+        Returns:
+            Cube of lightning data
+
+        Raises:
+            ValueError:
+                If one of the cubes is not found or doesn't match the other
+        """
+        cape, liftidx, pwat, cin, apcp = self._get_inputs(cubes)
+
+        # Regression equations require math on cubes with incompatible units, so strip data
+        templ = apcp.copy()
+        cape = cape.data
+        liftidx = liftidx.data
+        pwat = pwat.data
+        cin = cin.data
+        apcp = apcp.data / 1000 * 39.3701  # convert (kg m-3) to (inches)
```

Comment:
Yes, Ben raised the same comment above with regard to checking for units.  I don't necessarily want to convert the units within the code itself, especially if an input cube has completely incompatible units for some reason it may raise other issues that need to be resolved. However, I agree we should definitely check the units to make sure they're what we expect for the regression statistics to be applicable. 

We have added the following check in the `_get_inputs()` which does as you (and Ben) suggest, and I've also added a unit test to make sure this works: 

```
        output_cubes = iris.cube.CubeList()
        input_names = {
            "atmosphere_specific_convective_available_potential_energy": ["J kg-1"],
            "temperature_difference_between_ambient_air_and_air_lifted_adiabatically": ["K"],
            "precipitable_water": ["kg m-2", "mm"],
            "atmosphere_specific_convective_inhibition": ["J kg-1"],
            "precipitation_amount": ["kg m-2", "mm"],
        }

        for input_name, units in input_names.items():
            output_cubes.append(self._extract_input(cubes, input_name))
            if not output_cubes[-1].units in units:
                expected_unit_string = ' or '.join(map(str, units))
                received_unit_string = str(output_cubes[-1].units)
                raise ValueError(
                    f"The {output_cubes[-1].name()} units are incorrect, expected "
                    f"units as {expected_unit_string} but received {received_unit_string})."
```

Reply:
perfect, great to have these checks, thank you